pkg/daemon/ceph/client/radosnamespace.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util/exec"
+)
+
+// CreateRadosNamespace create a rados namespace in a pool.
+// poolName is the name of the ceph block pool, the same as the CephBlockPool CR name.
+func CreateRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	logger.Infof("creating rados namespace %q", namespaceName)
+	//  rbd namespace create pool-name/namespace-name
+	args := []string{"namespace", "create", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.EEXIST) {
+			logger.Infof("rados namespace %s/%s already exists", poolName, namespaceName)
+		}
+		return errors.Wrapf(err, "failed to create rados namespace %s/%s. %s", poolName, namespaceName, output)
+
+	}
+
+	logger.Infof("successfully created rados namespace %s/%s", poolName, namespaceName)
+	return nil
+}
+
+func getRadosNamespaceStatistics(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) (*PoolStatistics, error) {
+	var poolStats PoolStatistics
+
+	args := []string{"pool", "stats", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = true
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.ENOENT) {
+			return &poolStats, nil
+		}
+		return nil, errors.Wrapf(err, "failed to get pool stats. %s", string(output))
+	}
+	if err := json.Unmarshal(output, &poolStats); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal pool stats response")
+	}
+
+	return &poolStats, nil
+}
+
+func checkForImagesInRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	logger.Debugf("checking any images/snapshosts present in pool %s/%s", poolName, namespaceName)
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.

Here, this means we should identify the ceph cluser in which the pool is being deleted.